tt_metal/tt_stl/reflection.hpp

```diff
@@ -1087,11 +1152,32 @@ inline hash_t hash_object(const T& object) noexcept {
         if constexpr (DEBUG_HASH_OBJECT_FUNCTION) {
             fmt::print("Hashing std::set of type {}: {}\n", get_type_name<T>(), object);
         }
-        auto hash = 0;
+        hash_t hash = 0;
         for (const auto& element : object) {
             hash = hash_objects(hash, element);
         }
         return hash;
+    } else if constexpr (is_specialization_v<T, std::map>) {
+        if constexpr (DEBUG_HASH_OBJECT_FUNCTION) {
+            fmt::print("Hashing std::map of type {}: {}\n", get_type_name<T>(), object);
+        }
+        hash_t hash = 0;
+        for (const auto& [key, value] : object) {
+            hash = hash_objects(hash, key, value);
+        }
+        return hash;
+    } else if constexpr (is_specialization_v<T, std::unordered_map>) {
+        if constexpr (DEBUG_HASH_OBJECT_FUNCTION) {
+            fmt::print("Hashing std::unordered_map of type {}: {}\n", get_type_name<T>(), object);
+        }
+        constexpr hash_t seed = 0x9e3779b9;
+        hash_t hash = 0;
+        // Combine using xor so that the hash is order invariant
+        // Alternative could be to sort the keys before hashing
+        for (const auto& [key, value] : object) {
+            hash ^= hash_objects(seed, key, value);
```

Comment:
I would be a bit worried by this, possibly introduces some extra hash collisions. I would prefer to sort them instead

Reply:
That makes sense. I will change to sort instead

Reply:
@sminakov-tt I've updated to sort first then hash the same way as std::map. Let me know if this looks okay now.